tests/org.eclipse.emfcloud.modelserver.emf.tests/src/org/eclipse/emfcloud/modelserver/emf/AbstractResourceTest.java

```diff
@@ -21,12 +26,23 @@
 import org.eclipse.emfcloud.modelserver.common.codecs.EMFJsonConverter;
 import org.junit.After;
 import org.junit.Before;
+import org.junit.BeforeClass;
 
 public abstract class AbstractResourceTest {
-   public static final String RESOURCE_PATH = "resources/";
+   public static final String TEST_RESOURCES_PATH = "resources/";
+   public static final String RESOURCE_PATH = ".temp/";
    @SuppressWarnings({ "checkstyle:VisibilityModifier" })
    protected ResourceSetImpl resourceSet; // needed in ResourceManagerTest.java
 
+   @BeforeClass
+   public static void setupTestResources() throws IOException {
+      // copy test resources to a temporary resource location to avoid
+      // git changes if executing tests due to saving resources
+      File sourceDirectory = new File(TEST_RESOURCES_PATH);
+      File destinationDirectory = new File(RESOURCE_PATH);
+      copyDirectoryFiles(sourceDirectory, destinationDirectory);
```

Comment:
on a second thought why not use the `Files` api of nio ?
see here: https://www.baeldung.com/java-copy-file#nio2

Reply:
Thanks for the hint, I used now `FileUtils.copyDirectory` from the apache commons lib to cleanup the custom code.